02-express-tutorial/app.js

```diff
@@ -1 +1,67 @@
-console.log('Express Tutorial')
+//console.log('Express Tutorial')
+const express = require("express")
+const app = express()
+const PORT = 3000
+const { products } = require("./data");
+
+app.use(express.static("./public"))
+
+app.get("/api/v1/test", (req, res) => {
+    res.status(200).json({ message: "It worked!" })
+})
+
+app.get("/api/v1/products", (req, res) => {
+    res.status(200).json(products)
+})
+
+app.get("/api/v1/products/:productID",(req, res) => {
+    //res.json(req.params)
+    const idToFind = parseInt(req.params.productID);
+    const product = products.find((p) => p.id === idToFind);
+   console.log(product)
+    if (product === undefined) res.status(404).json( { message: "That product was not found."})
```

Comment:
Take a look at the http error code definitions for 204 and 404. I would return a 204 in this instance. 404 usually means "resource not found" as in that url is pointing somewhere that does not exist in the api.  In general 400 errors all mean that there was something broken on the client side of the request.

In this case the URL is fine and product lookup succeeded. It just didn't find a product. In essence it "Failed successfully" so I would return a 204.

However there are a lot of real world APIs out there right now returning a 204 in similar cases. so you're not wrong.